--- plugins/org.python.pydev/src/org/python/pydev/editor/PyEdit.java.ori	2017-01-17 10:45:29.395742259 +0000
+++ plugins/org.python.pydev/src/org/python/pydev/editor/PyEdit.java	2017-01-17 10:45:34.882678840 +0000
@@ -357,9 +357,6 @@
 
             CheckDefaultPreferencesDialog.askAboutSettings();
 
-            //Ask for people to consider funding PyDev.
-            PydevShowBrowserMessage.show();
-
             //Warn about the Eclipse version we require.
             checkEclipseRunning();
         } catch (Throwable e) {
